usb/imx-usbc/desc_manager.c

```diff
@@ -216,27 +219,35 @@ static int desc_ReqGetConfig(const usb_setup_packet_t *setup)
 
 static void desc_ReqGetDescriptor(const usb_setup_packet_t *setup)
 {
+	usb_string_desc_t *strDesc;
+
 	if (setup->wValue >> 8 == USB_DESC_DEVICE) {
 		ctrl_execTransfer(0, desc_common.dev, sizeof(usb_device_desc_t), USB_ENDPT_DIR_IN);
 	}
 	else if (setup->wValue >> 8 == USB_DESC_CONFIG) {
 		ctrl_execTransfer(0, desc_common.cfg, setup->wLength, USB_ENDPT_DIR_IN);
 	}
 	else if (setup->wValue >> 8 == USB_DESC_STRING) {
-		if ((setup->wValue & 0xff) == 0)
-			ctrl_execTransfer(0, desc_common.str_0, MIN(sizeof(usb_string_desc_t), setup->wLength), USB_ENDPT_DIR_IN);
-		else if ((setup->wValue & 0xff) == 1)
-			ctrl_execTransfer(0, desc_common.str_man, MIN(56, setup->wLength), USB_ENDPT_DIR_IN);
-		else if ((setup->wValue & 0xff) == 2)
-			ctrl_execTransfer(0, desc_common.str_prod, MIN(30, setup->wLength), USB_ENDPT_DIR_IN);
+		if ((setup->wValue & 0xff) == 0) {
+			strDesc = (usb_string_desc_t *)desc_common.str0;
+			ctrl_execTransfer(0, desc_common.str0, MIN(strDesc->bLength, setup->wLength), USB_ENDPT_DIR_IN);
+		}
+		else if ((setup->wValue & 0xff) == 1) {
+			strDesc = (usb_string_desc_t *)desc_common.strMan;
+			ctrl_execTransfer(0, desc_common.strMan, MIN(strDesc->bLength, setup->wLength), USB_ENDPT_DIR_IN);
+		}
+		else if ((setup->wValue & 0xff) == 2) {
+			strDesc = (usb_string_desc_t *)desc_common.strProd;
```

Comment:
In targets with MMU it doesn't work. `desc_common.str0`, `desc_common.strMan` and `desc_common.strProd` point to physical addresses. The best solution would be keep virtual addresses and make translation to physical address while transfer is executed.

Reply:
That's right, I definitely need to fix this.

Reply:
Because `va2pa()` cannot be used in the interrupt handler, I used an intermediate container with a physical address and a virtual address as a pointer to usb_string_desc_t.